### PR TITLE
enhance parsing model response function for more edge cases

### PR DESCRIPTION
**Pull request by ylwu-amzn**
### Description
Enhance parsing model response function for more edge cases
1. Model output is invalid json 
2. Model output has no valid field.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ public void createInteraction(
      * @param actionListener get all the final interactions that are not traces
      */
     public void getFinalInteractions(String conversationId, int lastNInteraction, ActionListener<List<Interaction>> actionListener) {
-        Preconditions.checkArgument(lastNInteraction > 0, "History message size must be at least 1.");
+        // Preconditions.checkArgument(lastNInteraction > 0, "History message size must be at least 1.");
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java:142]: are we going to keep it like this?

**Review by dhrubo-os — DISMISSED** [review]

**Review by dhrubo-os — APPROVED** [review]

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java (hunk 1 above)
Comment (ylwu-amzn) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java:142]: Yes, this is for allowing user not use conversation history if they want

Comment (dhrubo-os) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java:142]: I mean will we keep this as commented out?

Comment (ylwu-amzn) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java]: fixed in new commit

**Review by b4sjoo — APPROVED** [review]

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**